### PR TITLE
Fixes #186 Extend schema

### LINKED ISSUE #186: Extend schema
* In DDI-Ratio and PK-Ratio plots: allow to use `PKParameters `field instead of `PKParameter `(s. https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/issues/655)
* In DDI-Ratio plots: add `Subunits `field (s. https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/issues/563)

### PR DESCRIPTION
**Pull request by Yuri05**

### CODE REVIEW

File: schemas/OSP_Qualification_Plan_Schema.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
         "Protocol",
         "Event",
         "Formulation",
-        "ObserverSet"
+        "ObserverSet",
+        "Expression"
```
Comment (Yuri05) [inline schemas/OSP_Qualification_Plan_Schema.json:15]: Just added "Expression" for the future use of the expression profiles BB.
Not sure if it should be expression or "Protein Expression" or "Expression Profile", but we can quickly rename if required.

File: schemas/OSP_Qualification_Plan_Schema.json [hunk 2]
```diff
@@ -571,6 +569,61 @@
...
+        "type": "string",
+        "enum": [
+          "Mechanism",
+          "Perpetrator",
+          "Victim"
+        ]
+      },
+      "minItems": 1,
+      "uniqueItems": true
+    },
```
Comment (Yuri05) [inline schemas/OSP_Qualification_Plan_Schema.json:585]: Subunits definition for DDI plots as discussed in https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/issues/563

File: schemas/OSP_Qualification_Plan_Schema.json [hunk 3]
```diff
@@ -1013,25 +1068,40 @@
...
                 ]
               },
               {
                 "required": [
-                  "SectionId",
-                  "Title",
                   "PKParameter",
-                  "PlotTypes",
-                  "Groups"
+                  "PlotType"
```
Comment (Yuri05) [inline schemas/OSP_Qualification_Plan_Schema.json:1104]: Finally found the way to use always required fields only once. So here `SectionId`, `Title `and `Groups `are always required. And then exactly one field from {`PKParameters`, `PKParameter`} and one from {`PlotTypes`, `PlotType`}
```
           "required": [
              "SectionId",
              "Title",
              "Groups"
            ],
            "anyOf": [
              {
                "required": [
                  "PKParameters",
                  "PlotTypes"
                ]
              },
              {
                "required": [
                  "PKParameter",
                  "PlotTypes"
                ]
              },
              {
                "required": [
                  "PKParameters",
                  "PlotType"
                ]
              },
              {
                "required": [
                  "PKParameter",
                  "PlotType"
                ]
              }
            ],
```

File: schemas/OSP_Qualification_Plan_Schema.json [hunk 4]
```diff
@@ -1127,9 +1200,20 @@
...
+                "required": [
+                  "PKParameters"
+                ]
+              },
+              {
+                "required": [
+                  "PKParameter"
+                ]
+              }
+            ],
```
Comment (Yuri05) [inline schemas/OSP_Qualification_Plan_Schema.json:1216]: same for the required fields here

File: schemas/OSP_Qualification_Plan_Schema.json (hunk 1 above)
Comment (msevestre) [reply schemas/OSP_Qualification_Plan_Schema.json:15]: ExpressionProfile. It's using the enum in PKSim so it needs to be verbatim

File: schemas/OSP_Qualification_Plan_Schema.json (hunk 3 above)
Comment (msevestre) [reply schemas/OSP_Qualification_Plan_Schema.json:1104]: Nice!

**Review by msevestre — CHANGES_REQUESTED** [review]
Name of constant needs to be changed

File: schemas/OSP_Qualification_Plan_Schema.json (hunk 1 above)
Comment (Yuri05) [reply schemas/OSP_Qualification_Plan_Schema.json]: done

**Pull request merged**